Source/Test/Rewriting/RewritingEngine.cs

```diff
@@ -52,6 +52,11 @@ public class RewritingEngine
         /// </summary>
         private readonly Regex DisallowedAssemblies;
 
+        /// <summary>
+        /// Simple cache to reduce redundant warnings.
+        /// </summary>
+        private readonly HashSet<string> ResolveWarnings = new HashSet<string>();
```

Comment:
This is awesome!

Reply:
Yeah, I was getting a million copies of the same warning which was a bit annoying...